src/program/l2fwd/l2fwd.lua

```diff
@@ -0,0 +1,84 @@
+module(...,package.seeall)
+
+local lib = require("core.lib")
+local main = require("core.main")
+
+L2Fwd = {}
+
+function L2Fwd.new(conf)
+   local o = {}
+   return setmetatable(o, { __index = L2Fwd })
+end
+
+function L2Fwd:push()
+   local input, output = assert(self.input.input), assert(self.output.output)
+
+   while not link.empty(input) do
+      link.transmit(output, link.receive(input))
```

Comment:
The DPDK L2FWD app actually does some processingon each packet. See here: http://dpdk.org/browse/dpdk/tree/examples/l2fwd/main.c#n239

The main idea is that if I send traffic with a destination MAC A and I don't change it in the L2FWD, then all possible switches between the sender and the L2FWD will get messed. Like if we use a VMDq port then the integrated switch will learn the MAC A on a ceratin port and won't forward the returned packet to the external port.

So what DPDK does is to replace each packet's destination MAC address with `02:00:00:00:00:xx`, where `xx` is the number for the port it came from (DPDK actually assigns numbers to each port it recognizes). And then also changes the source with the MAC of the port it send it through. So it is more or less a real forwarding.

But SnabbSwitch is in an even better position since it actually has a real switching app - the learning bridge. I guess it is the most suitable in this situation.